packages/framework/src/client/scheduler.js

```diff
@@ -1,109 +1,148 @@
-import { find } from 'lodash';
+import { find, isEmpty, isEqual, isNil, uniq } from 'lodash';
 import ResourceRequest, { STATUS } from './resource-request';
 
-const DEFAULT_FETCH_INTERVAL = 5000; // Five seconds
+const DEFAULT_TIMER_INTERVAL = 100; // 1/10 second response time
+const DEFAULT_READ_OPERATION = 'read';
 
 export default class Scheduler {
+	/**
+	 * Creates a new Scheduler
+	 * @param {Object} operations The mapping of operation names to functions
+	 * @param {number} timerInterval Milliseconds between timer iterations (defaults to 500)
+	 * @param {function} setInterval The function to set an interval timer (defaults to window.setInterval)
+	 * @param {function} clearInterval The function to clear an internal timer by id (defaults to window.setInterval)
+	 */
 	constructor(
-		fetch,
-		fetchInterval = DEFAULT_FETCH_INTERVAL,
+		operations,
+		timerInterval = DEFAULT_TIMER_INTERVAL,
 		setInterval = window.setInterval,
+		clearInterval = window.clearInterval,
 	) {
-		this.fetch = fetch;
+		this.operations = operations;
 		this.requests = [];
 		this.timerId = null;
 
-		if ( fetchInterval > 0 ) {
-			setInterval( () => {
+		if ( timerInterval > 0 ) {
+			this.timerId = setInterval( () => {
 				this.cleanUp();
 				this.resendTimeouts();
 				this.sendReadyRequests();
-			}, fetchInterval );
+			}, timerInterval );
 		}
+
+		/**
+		 * Stops the timer for this scheduler (if running)
+		 */
+		this.stopTimer = () => {
```

Comment:
The `stopTimer` method should be defined like all the other methods of this class, for consistency, instead of creating this one method inside the constructor.

Reply:
The reason I was declaring it here, is so that I don't have to hold a reference to `clearInterval` anywhere else. I was trying to isolate `setInterval` and `clearInterval` to only the constructor this way. (granted, this will change to `setTimeout` and `clearTimeout` anyway)

Reply:
Outdated, new implementation of stop makes it a normal member function.